DependencyInjection/HttplugExtension.php

```diff
@@ -207,4 +217,20 @@ private function configureAuthentication(ContainerBuilder $container, array $con
                 ->addArgument(new Reference($authServiceKey));
         }
     }
+
+    /**
+     * Verifies that Puli bundle is installed.
+     *
+     * @param ContainerBuilder $container
+     *
+     * @throws \Exception
+     */
+    private function verifyPuliInstalled(ContainerBuilder $container)
+    {
+        if (false === $container->has('puli.discovery')) {
```

Comment:
What if we do the following: remove the fallback services and move them to a compiler pass. Then we can check in the compiler if all the services has been defined in the extension or not? (This requires to know in what order are the compilers and extensions are loaded). If any of the services are not registered, register it with the factory (and register the factory as well).